lib/alp_workloads/kvm_workload_utils.pm

```diff
@@ -0,0 +1,201 @@
+# SUSE's openQA tests
+#
+# Copyright 2022 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: This file provides the utility functions for kvm workload container.
+# Maintainer: alice <xlai@suse.com>, qe-virt@suse.de
+
+package alp_workloads::kvm_workload_utils;
+
+use base Exporter;
+use Exporter;
+use strict;
+use warnings;
+use testapi;
+use Utils::Architectures;
+use Utils::Backends qw(use_ssh_serial_console);
+use ipmi_backend_utils;
+use utils;
+use virt_autotest::virtual_network_utils qw(clean_all_virt_networks);
+
+our @EXPORT = qw(
+  set_ct_engine
+  set_kvm_image
+  pull_kvm_container
+  config_host_and_kvm_container
+  cleanup_host_and_kvm_container
+  start_kvm_container
+  enter_kvm_container_sh
+  exit_kvm_container
+  install_tools_within_kvm_container
+  setup_services_within_kvm_container
+  collect_kvm_container_setup_logs
+  setup_kvm_container_from_scratch
+  clean_and_resetup_kvm_container
+);
+
+# Container engine: can be podman(default), docker(not tested)
+our $ct_engine = "podman";
+
+my $DEFAULT_KVM_IMAGE = "registry.opensuse.org/suse/alp/workloads/tumbleweed_containerfiles/suse/alp/workloads/kvm:latest";
+our $kvm_image = $DEFAULT_KVM_IMAGE;
+
+sub set_ct_engine {
+    $ct_engine = shift;
+
+    die "Container engine can only be set to podman or docker!" if ($ct_engine ne "podman" && $ct_engine ne "docker");
+}
+
+sub set_kvm_image {
+    my $_image = shift;
+
+    # Allow customization in case of special test needs
+    if ($_image ne $DEFAULT_KVM_IMAGE) {
+        $kvm_image = $_image;
+        record_info("The kvm container image has been changed from default!", "Now test is using kvm image: $kvm_image.");
+    }
+}
+
+sub pull_kvm_container {
```

Comment:
I think
```
pull_kvm_container_image
```
is better.

Reply:
Generally speaking, I think your function names should differentiate "image" from "container".

Reply:
Will use your suggested name.

I had a double check, set_kvm_image and this function are the only two functions related with image. After updating this one, all will be fine.

Reply:
done